bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/DeviceThingHandler.java

```diff
@@ -0,0 +1,541 @@
+                        }
+                    }, 3, TimeUnit.SECONDS);
+                }
+            }
+            return;
+        }
+
+        Resource newResource;
+        switch (channelUID.getId()) {
+            case HueBindingConstants.CHANNEL_COLORTEMPERATURE:
+                newResource = new Resource(ResourceType.LIGHT).setColorTemperaturePercent(command,
+                        mirekSchemaFrom(ResourceType.LIGHT));
+                break;
+
+            case HueBindingConstants.CHANNEL_COLORTEMPERATURE_ABS:
+                newResource = new Resource(ResourceType.LIGHT).setColorTemperatureKelvin(command);
+                break;
+
+            case HueBindingConstants.CHANNEL_COLOR:
+                newResource = new Resource(ResourceType.LIGHT).setColor(command);
+                break;
+
+            case HueBindingConstants.CHANNEL_BRIGHTNESS:
+                newResource = new Resource(ResourceType.LIGHT).setBrightness(command);
+                break;
+
+            case HueBindingConstants.CHANNEL_SWITCH:
+                newResource = new Resource(ResourceType.LIGHT).setSwitch(command);
+                break;
+
+            case HueBindingConstants.CHANNEL_TEMPERATURE_ENABLED:
+                newResource = new Resource(ResourceType.TEMPERATURE).setEnabled(command);
+                break;
+
+            case HueBindingConstants.CHANNEL_MOTION_ENABLED:
+                newResource = new Resource(ResourceType.MOTION).setEnabled(command);
+                break;
+
+            case HueBindingConstants.CHANNEL_LIGHT_LEVEL_ENABLED:
+                newResource = new Resource(ResourceType.LIGHT_LEVEL).setEnabled(command);
+                break;
+
+            default:
+                return; // <= nota bene !!
+        }
+
+        String resourceId = commandResourceIds.get(newResource.getType());
+        if (Objects.nonNull(resourceId)) {
+            try {
+                getBridgeHandler().putResource(newResource.setId(resourceId));
+            } catch (ApiException | AssetNotLoadedException e) {
+                logger.warn("handleCommand() error {}", e.getMessage(), e);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initialize() called");
+        DeviceConfig config = getConfigAs(DeviceConfig.class);
+
+        String resourceId = config.resourceId;
+        if (Objects.isNull(resourceId) || resourceId.isEmpty()) {
+            logger.debug("initialize() configuration resourceId is bad");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.clip2.conf-error-resource-id-bad");
+            return;
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        thisResource.setId(resourceId);
+
+        supportedChannelIds.clear();
+        commandResourceIds.clear();
+        contributorsCache.clear();
+        controlIds.clear();
```

Comment:
Perhaps this should be done in `dispose()` instead?

Reply:
It is in `initialize()` to cover the edge case that the core might call `initialize()` again on an already initialized handler, without first calling `dispose()` on it.

Reply:
This would be a bug in core and would also cause a lot of wild running background threads when reinitializing handlers. We should be able to trust the framework's lifecycle management. Instead these caches will now not be cleared when handler is disposed.

Reply:
Ok.